.github/workflows/upload_nightly_wheels.yml

```diff
@@ -0,0 +1,65 @@
+name: Upload nightly wheels to Anaconda Cloud
+
+on:
+  # Run daily at 1:23 UTC to upload nightly wheels to Anaconda Cloud
+  schedule:
+    - cron: '23 1 * * *'
+  # Run on demand with workflow dispatch
+  workflow_dispatch:
+
+permissions:
+  actions: read
+
+jobs:
+  upload_nightly_wheels:
+    name: Upload nightly wheels to Anaconda Cloud
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        # The login shell is necessary for the setup-micromamba setup
+        # to work in subsequent jobs.
+        # https://github.com/mamba-org/setup-micromamba#about-login-shells
+        shell: bash -e -l {0}
+    if: github.repository_owner == 'scikit-hep'
+
+    steps:
+      # https://github.com/actions/download-artifact/issues/3#issuecomment-1017141067
+      - name: Download wheel artifacts from last build on 'main'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          PROJECT_REPO="scikit-hep/awkward"
+          BRANCH="main"
+          WORKFLOW_NAME="packaging-test.yml"
+          ARTIFACT_PATTERN="awkward*wheel*"  # awkward-wheel and awkward-cpp-wheels-*
+
+          gh run --repo "${PROJECT_REPO}" \
+             list --branch "${BRANCH}" \
+                  --workflow "${WORKFLOW_NAME}" \
+                  --json event,status,conclusion,databaseId > runs.json
+          RUN_ID=$(
+            jq --compact-output \
+              '[
+                .[] |
+                # Filter on "push" events to main (merged PRs) ...
+                select(.event == "push") |
```

Comment:
Now selecting the `push` events that were on `main`, which comes from runs set off by merged PRs.